In [1]:
# Import libraries
import pandas as pd
import numpy as np
import os
import matplotlib.pyplot as plt
import seaborn as sns
import scipy


In [2]:
# Define path
path = r'/Users/Shikongo/Healthcare Cost Analysis and Prediction/'

In [3]:
path

'/Users/Shikongo/Healthcare Cost Analysis and Prediction/'

In [4]:
df_insurance = pd.read_csv(r'/Users/Shikongo/Healthcare Cost Analysis and Prediction/Healthcare Raw Data/Raw data/insurance.csv',index_col = False)

In [5]:
df_insurance = pd.read_csv(os.path.join(path,'Healthcare Raw Data','Raw data','insurance.csv'),index_col = False)

1 Data Wrangling

In [6]:
# No column to be dropped from insuarance data

In [7]:
df_insurance.head()

Unnamed: 0  Age     Sex     BMI  Children Smoker     Region        Charges
0           0   19  female  27.900         0    yes  southwest  $16884.924000
1           1   18    male  33.770         1     no  southeast   $1725.552300
2           2   28    male  33.000         3     no  southeast   $4449.462000
3           3   33    male  22.705         0     no  northwest  $21984.470610
4           4   32    male  28.880         0     no  northwest   $3866.855200

In [8]:
# Rename columns: Change variables started with capital letter rather than small letter. And also capitalised BMI.
df_insurance.rename(columns={'age':'Age','sex':'Sex','bmi':'BMI','children':'Children','smoker':'Smoker','region':'Region','charges':'Charges'},inplace =True)

In [9]:
df_insurance.head()

Unnamed: 0  Age     Sex     BMI  Children Smoker     Region        Charges
0           0   19  female  27.900         0    yes  southwest  $16884.924000
1           1   18    male  33.770         1     no  southeast   $1725.552300
2           2   28    male  33.000         3     no  southeast   $4449.462000
3           3   33    male  22.705         0     no  northwest  $21984.470610
4           4   32    male  28.880         0     no  northwest   $3866.855200

In [10]:
# Check mixed data type: No mixed data type found
for col in df_insurance.columns.tolist():
  weird = (df_insurance[[col]].applymap(type) != df_insurance[[col]].iloc[0].apply(type)).any(axis = 1)
  if len (df_insurance[weird]) > 0:
    print (col)

In [11]:
# Check missing values: No missing values found
df_insurance.isnull().sum()

Unnamed: 0    0
Age           0
Sex           0
BMI           0
Children      0
Smoker        0
Region        0
Charges       0
dtype: int64

In [12]:
# Find duplicates: No duplicate
df_insurance_clean = df_insurance

In [13]:
df_dups = df_insurance_clean[df_insurance_clean.duplicated()]

In [14]:
# Descriptive statistical analysis
df_insurance.describe()

Unnamed: 0          Age          BMI     Children
count  1338.000000  1338.000000  1338.000000  1338.000000
mean    668.500000    39.207025    30.663397     1.094918
std     386.391641    14.049960     6.098187     1.205493
min       0.000000    18.000000    15.960000     0.000000
25%     334.250000    27.000000    26.296250     0.000000
50%     668.500000    39.000000    30.400000     1.000000
75%    1002.750000    51.000000    34.693750     2.000000
max    1337.000000    64.000000    53.130000     5.000000

Interprent above output
1.	Age:
The minimum age is 18, and the maximum age is 64. These values seem reasonable and within a typical range for adults.
2.	BMI (Body Mass Index):
The minimum BMI is 15.96, and the maximum BMI is 53.13. These values also appear to be within a feasible range for BMI.
3.	Children:
The minimum number of children is 0, and the maximum is 5. This seems appropriate, as it's common for people to have between 0 and 5 children.
4.	Charges:
The minimum insurance charge is $1,121.87, and the maximum is $63,770.43. These values show a wide range of insurance charges, which is not unusual in healthcare.

Overall, based on the minimum and maximum values, and  averages seem within expected ranges and nothing immediately appears to be off or suspicious in the data.  


In [15]:
df_insurance.to_csv(os.path.join(path,'Healthcare Clean Data',' Clean Data','insurance2.csv'))